pyuvdata/uvflag/uvflag.py

```diff
@@ -667,6 +682,7 @@ def _set_type_waterfall(self):
         self._Nants_data.required = False
         self._Nbls.required = False
         self._Nspws.required = False
+        self._weights_square_array.required = False
```

Comment:
I'm not sure this line is necessary. It's initialized as not required and I don't see anywhere that `_weights_square_array.required` is ever set to True.

Reply:
Oops! I meant to remove this one. I'll remove it in my next push after I write a round trip test.